### PR TITLE
VOF boundary conditions

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
# Description of the problem

When solving different cases, boundary conditions were only applied to the N-S equations. Therefore, when applying boundary conditions such as a periodic boundary condition, the right fluid was not necessarily transported on the opposite side of the domain. In the same way, until now, it was not possible to specify which fluid would enter through an inlet or exit through an outlet.

# Description of the solution

- Periodic boundary condition for the phase fraction was implemented implicitly in the code when a periodic boundary condition is specified in the ``boundary conditions`` subsection of the parameter file.
- Dirichlet boundary condition for the phase fraction was implemented explicitly with the new ``dirichlet`` type in the ``boundary conditions VOF`` subsection of the parameter file.

# How Has This Been Tested?

Two tests were added to test the two types of boundary conditions.

- [x] applications_tests/gls_navier_stokes_2d/gls_vof_periodic_boundary_condition
- [x] applications_tests/gls_navier_stokes_2d/gls_vof_dirichlet_boundary_condition

# Documentation

In the multiphysics boundary conditions' documentation, the new boundary conditions were added.

- [x] doc/source/parameters/cfd/boundary_conditions_multiphysics.rst

# Comments

- A gravity wave example folder was also added to multiphysics examples. This example will be completed after further improvements of the VOF model.

### CODE REVIEW

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst [hunk 1]
```diff
@@ -87,20 +87,27 @@ For tracer boundary conditions, the defaults parameters are:
...
+            set id 		= 0
             set type	        = none
         end
         subsection bc 1
-	    set id 		= 1
+            set id 		= 1
             set type		= peeling/wetting
         end
+        subsection bc 2
+            set id 		= 3
```
Comment (blaisb) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:105]: Maybe set id to 2 here?

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst [hunk 2]
```diff
@@ -87,20 +87,27 @@ For tracer boundary conditions, the defaults parameters are:
 VOF
 ^^^
 
-For VOF boundary conditions (multiphase flow), the possible ``types`` are ``none`` (default) and ``peeling/wetting``, as shown below.
+For VOF boundary conditions (multiphase flow), the possible ``types`` are ``none`` (default), ``peeling/wetting``, and ``dirichlet``, as shown below.
```
Comment (blaisb) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:90]:
```suggestion
For VOF boundary conditions (multiphase flow), the possible ``types`` are ``none`` (default, which corresponds to a zero normal flux), ``peeling/wetting``, and ``dirichlet``, as shown below.
```

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst [hunk 3]
```diff
@@ -118,5 +125,8 @@ For VOF boundary conditions (multiphase flow), the possible ``types`` are ``none
 * ``type``: This is the type of boundary condition been imposed. At the moment, choices are:
     * ``none`` for which nothing happens.
     * ``peeling/wetting`` for the fluid can attach to (`wet`) or detach from (`peel`) the boundary. The parameters for peeling/wetting are defined in the :doc:`volume_of_fluid` subsection of the parameter file.
+    * ``dirichlet`` for inlet and outlet boundary conditions, to specify which fluid should be at the selected boundary.
 
+.. note::
```
Comment (blaisb) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:130]: Cool :)!

**Review by blaisb — CHANGES_REQUESTED** [review]
1 or 2 tiny change. Very very cool PR.

**Review by blaisb — APPROVED** [review]

File: examples/multiphysics/sloshing-in-rectanglular-tank/sloshing_020/sloshing_post_processing.py
Comment (hepap) [inline examples/multiphysics/sloshing-in-rectanglular-tank/sloshing_020/sloshing_post_processing.py:1]: The post processing script is duplicated I think. Maybe you can put only once in the examples `sloshing-in-rectanglular-tank` folder :)

**Review by hepap — APPROVED** [review]
Really nice PR :D good work!
